android/src/main/java/io/ably/flutter/plugin/AblyMessageCodec.java

```diff
@@ -356,25 +369,31 @@ private TokenDetails decodeTokenDetails(Map<String, Object> jsonMap) {
 
   private ChannelOptions decodeRestChannelOptions(Map<String, Object> jsonMap) {
     if (jsonMap == null) return null;
-    final Object cipher = jsonMap.get(PlatformConstants.TxRealtimeChannelOptions.cipher);
-    try {
-      return createChannelOptions(cipher);
-    } catch (AblyException e) {
-      System.out.println("Exception while decoding RestChannelOptions: " + e);
-      return null;
+    ChannelOptions options = new ChannelOptions();
+    options.cipherParams = decodeCipherParams((Map<String, Object>) jsonMap.get(PlatformConstants.TxRestChannelOptions.cipherParams));
```

Comment:
Very safe because I wrote the code on the other side. 
https://github.com/ably/ably-flutter/blob/0898c07cf4487b8b8614260ad28daa11e64f336d/lib/src/platform/src/codec.dart#L379-L390

This is the way Flutter plugins pass information back and forth from Dart and Java side. You can see this pattern all over Ably Flutter, but also every package plugin (A flutter plugin which contains iOS or Android code).

For example, look at this implementation in the Flutterfire firebase_messaging library: https://github.com/FirebaseExtended/flutterfire/blob/a91fa459add243b37a7da9a67fbe94f2116bb3ca/packages/cloud_firestore/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/firestore/FlutterFirebaseFirestorePlugin.java#L271-L289

Although we cannot **statically** guarantee the type is correct before casting, **checking** the type at runtime and throwing an error or logging if it is not as expected adds verbosity, complexity with no additional benefits IMHO.

Reply:
Thanks for clarification 👍